frontend/src/metabase-lib/expression.ts

```diff
@@ -73,7 +74,7 @@ export function expressionClauseForLegacyExpression(
 export function legacyExpressionForExpressionClause(
   query: Query,
   stageIndex: number,
-  expressionClause: ExpressionClause,
+  expressionClause: ExpressionClause | AggregationClause,
```

Comment:
We need to support `FilterClause` here as well

Reply:
Added `FilterClause` and a test.